* Data Loading


In [7]:
import pandas as pd
import warnings
warnings.filterwarnings("ignore")

In [8]:
df=pd.read_csv(r"C:\Users\dell\Documents\Data Analysis files\Amazon_sales\data\Amazon Sale Report.csv")

In [9]:
df.head()

,index,Order ID,Date,Status,Fulfilment,Sales Channel,ship-service-level,Style,SKU,Category,...,currency,Amount,ship-city,ship-state,ship-postal-code,ship-country,promotion-ids,B2B,fulfilled-by,Unnamed: 22
0,94760,171-0243036-7059512,6/26/2022,Shipped,Amazon,Amazon.in,Expedited,SET389,SET389-KR-NP-S,Set,...,INR,648.00,VASAI VIRAR,MAHARASHTRA,401209.0,IN,IN Core Free Shipping 2015/04/08 23-48-5-108,False,NaN,False
1,1,171-9198151-1101146,4/30/2022,Shipped - Delivered to Buyer,Merchant,Amazon.in,Standard,JNE3781,JNE3781-KR-XXXL,kurta,...,INR,406.00,BENGALURU,KARNATAKA,560085.0,IN,Amazon PLCC Free-Financing Universal Merchant ...,False,Easy Ship,NaN
2,2,404-0687676-7273146,4/30/2022,Shipped,Amazon,Amazon.in,Expedited,JNE3371,JNE3371-KR-XL,kurta,...,INR,329.00,NAVI MUMBAI,MAHARASHTRA,410210.0,IN,IN Core Free Shipping 2015/04/08 23-48-5-108,True,NaN,NaN
3,3,403-9615377-8133951,4/30/2022,Cancelled,Merchant,Amazon.in,Standard,J0341,J0341-DR-L,Western Dress,...,INR,753.33,PUDUCHERRY,PUDUCHERRY,605008.0,IN,NaN,False,Easy Ship,NaN
4,4,407-1069790-7240320,4/30/2022,Shipped,Amazon,Amazon.in,Expedited,JNE3671,JNE3671-TU-XXXL,Top,...,INR,574.00,CHENNAI,TAMIL NADU,600073.0,IN,NaN,False,NaN,NaN


* Checking for null values

In [10]:
df.isnull().sum()

index                     0
Order ID                  0
Date                      0
Status                    0
Fulfilment                0
Sales Channel             0
ship-service-level        0
Style                     0
SKU                       0
Category                  0
Size                      0
ASIN                      0
Courier Status         6872
Qty                       0
currency               7795
Amount                 7795
ship-city                33
ship-state               33
ship-postal-code         33
ship-country             33
promotion-ids         49153
B2B                       0
fulfilled-by          89698
Unnamed: 22           49050
dtype: int64

In [11]:
df.shape

(128975, 24)

In [15]:
df.columns

Index(['index', 'Order ID', 'Date', 'Status', 'Fulfilment', 'Sales Channel ',
       'ship-service-level', 'Style', 'SKU', 'Category', 'Size', 'ASIN',
       'Courier Status', 'Qty', 'currency', 'Amount', 'ship-city',
       'ship-state', 'ship-postal-code', 'ship-country', 'promotion-ids',
       'B2B', 'fulfilled-by', 'Unnamed: 22'],
      dtype='object')

* drop column "fulfilled-by": missing values:> 89698
* drop column "promotion-ids": missing values:> 49153
* drop column "Unnamed: 22": missing values:> 49050


In [17]:
df=df.drop("Unnamed: 22",axis=1)
df=df.drop("promotion-ids",axis=1)
df=df.drop("fulfilled-by",axis=1)

In [19]:
df.describe()

,index,Qty,Amount,ship-postal-code
count,128975.000000,128975.000000,121180.000000,128942.000000
mean,64487.000000,0.904431,648.561465,463966.236509
std,37232.019822,0.313354,281.211687,191476.764941
min,0.000000,0.000000,0.000000,110001.000000
25%,32243.500000,1.000000,449.000000,382421.000000
50%,64487.000000,1.000000,605.000000,500033.000000
75%,96730.500000,1.000000,788.000000,600024.000000
max,128974.000000,15.000000,5584.000000,989898.000000


In [20]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 128975 entries, 0 to 128974
Data columns (total 21 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   index               128975 non-null  int64  
 1   Order ID            128975 non-null  object 
 2   Date                128975 non-null  object 
 3   Status              128975 non-null  object 
 4   Fulfilment          128975 non-null  object 
 5   Sales Channel       128975 non-null  object 
 6   ship-service-level  128975 non-null  object 
 7   Style               128975 non-null  object 
 8   SKU                 128975 non-null  object 
 9   Category            128975 non-null  object 
 10  Size                128975 non-null  object 
 11  ASIN                128975 non-null  object 
 12  Courier Status      122103 non-null  object 
 13  Qty                 128975 non-null  int64  
 14  currency            121180 non-null  object 
 15  Amount              121180 non-nul

In [21]:
df["currency"].value_counts()

currency
INR    121180
Name: count, dtype: int64

In [22]:
df=df.drop("currency",axis=1)

In [23]:
df["Courier Status"].value_counts()

Courier Status
Shipped      109487
Unshipped      6681
Cancelled      5935
Name: count, dtype: int64

* amount fill with median
* courier status fill with mode
* ship postal code fill with median
* [ship_city,ship_country,ship_state] fill with mode

In [24]:
 df["Amount"].fillna(df["Amount"].median(),inplace=True)

In [26]:
mode1=df['Courier Status'].mode()[0]

In [27]:
df["Courier Status"]=df["Courier Status"].fillna(mode1)

In [28]:
df["ship-postal-code"].fillna(df["ship-postal-code"].median(),inplace=True)

In [29]:
ship=df["ship-city"].mode()[0]
ship_c=df["ship-country"].mode()[0]
ship_s=df["ship-state"].mode()[0]

In [31]:
df["ship-city"]=df["ship-city"].fillna(ship)
df["ship-country"]=df["ship-country"].fillna(ship_c)
df["ship-state"]=df["ship-state"].fillna(ship_c)


In [32]:
df.isnull().sum()

index                 0
Order ID              0
Date                  0
Status                0
Fulfilment            0
Sales Channel         0
ship-service-level    0
Style                 0
SKU                   0
Category              0
Size                  0
ASIN                  0
Courier Status        0
Qty                   0
Amount                0
ship-city             0
ship-state            0
ship-postal-code      0
ship-country          0
B2B                   0
dtype: int64

In [33]:
df.head()

,index,Order ID,Date,Status,Fulfilment,Sales Channel,ship-service-level,Style,SKU,Category,Size,ASIN,Courier Status,Qty,Amount,ship-city,ship-state,ship-postal-code,ship-country,B2B
0,94760,171-0243036-7059512,6/26/2022,Shipped,Amazon,Amazon.in,Expedited,SET389,SET389-KR-NP-S,Set,S,B09KXVBD7Z,Shipped,1,648.00,VASAI VIRAR,MAHARASHTRA,401209.0,IN,False
1,1,171-9198151-1101146,4/30/2022,Shipped - Delivered to Buyer,Merchant,Amazon.in,Standard,JNE3781,JNE3781-KR-XXXL,kurta,3XL,B09K3WFS32,Shipped,1,406.00,BENGALURU,KARNATAKA,560085.0,IN,False
2,2,404-0687676-7273146,4/30/2022,Shipped,Amazon,Amazon.in,Expedited,JNE3371,JNE3371-KR-XL,kurta,XL,B07WV4JV4D,Shipped,1,329.00,NAVI MUMBAI,MAHARASHTRA,410210.0,IN,True
3,3,403-9615377-8133951,4/30/2022,Cancelled,Merchant,Amazon.in,Standard,J0341,J0341-DR-L,Western Dress,L,B099NRCT7B,Shipped,0,753.33,PUDUCHERRY,PUDUCHERRY,605008.0,IN,False
4,4,407-1069790-7240320,4/30/2022,Shipped,Amazon,Amazon.in,Expedited,JNE3671,JNE3671-TU-XXXL,Top,3XL,B098714BZP,Shipped,1,574.00,CHENNAI,TAMIL NADU,600073.0,IN,False


* Inconsistent formats:
* 
 -Standardized category names
 -replace rajshthan to rajasthan , rj to rajasthan

In [34]:
df["ship-state"].value_counts()

ship-state
MAHARASHTRA      22260
KARNATAKA        17326
TAMIL NADU       11483
TELANGANA        11330
UTTAR PRADESH    10638
                 ...  
Nagaland             1
Mizoram              1
bihar                1
rajsthan             1
Rajsthan             1
Name: count, Length: 70, dtype: int64

In [35]:
df["ship-state"]=df["ship-state"].str.lower()

In [36]:
df['ship-state']=df["ship-state"].replace("rajshthan","rajasthan")


In [37]:
df['ship-state']=df["ship-state"].replace("rajsthan","rajasthan")


In [38]:
df['ship-state']=df["ship-state"].replace("rj","rajasthan")


In [39]:
df['ship-state']=df["ship-state"].replace("punjab/mohali/zirakpur","punjab")


In [40]:
df['ship-state']=df["ship-state"].replace("pb","punjab")


In [41]:
df['ship-state']=df["ship-state"].replace("nl","nagaland")


In [42]:
df['ship-state']=df["ship-state"].replace("ar","arunachal pradesh")


In [43]:
df['ship-state']=df["ship-state"].replace("apo","Unknown")


In [44]:
df['ship-state']=df["ship-state"].replace("in","Unknown")

In [45]:
df["ship-state"]=df["ship-state"].str.replace(" ","_")

In [47]:
df["ship-state"].value_counts()

ship-state
maharashtra           22260
karnataka             17326
tamil_nadu            11483
telangana             11330
uttar_pradesh         10638
delhi                  6967
kerala                 6585
west_bengal            5963
andhra_pradesh         5430
gujarat                4489
haryana                4415
rajasthan              2718
madhya_pradesh         2529
odisha                 2136
bihar                  2114
punjab                 1919
assam                  1663
uttarakhand            1553
jharkhand              1456
goa                    1137
chhattisgarh            909
himachal_pradesh        788
jammu_&_kashmir         702
puducherry              350
chandigarh              333
manipur                 316
andaman_&_nicobar_      257
meghalaya               207
sikkim                  205
nagaland                187
tripura                 151
arunachal_pradesh       147
new_delhi                81
mizoram                  76
dadra_and_nagar          70
ladakh   

In [48]:
df.head()

,index,Order ID,Date,Status,Fulfilment,Sales Channel,ship-service-level,Style,SKU,Category,Size,ASIN,Courier Status,Qty,Amount,ship-city,ship-state,ship-postal-code,ship-country,B2B
0,94760,171-0243036-7059512,6/26/2022,Shipped,Amazon,Amazon.in,Expedited,SET389,SET389-KR-NP-S,Set,S,B09KXVBD7Z,Shipped,1,648.00,VASAI VIRAR,maharashtra,401209.0,IN,False
1,1,171-9198151-1101146,4/30/2022,Shipped - Delivered to Buyer,Merchant,Amazon.in,Standard,JNE3781,JNE3781-KR-XXXL,kurta,3XL,B09K3WFS32,Shipped,1,406.00,BENGALURU,karnataka,560085.0,IN,False
2,2,404-0687676-7273146,4/30/2022,Shipped,Amazon,Amazon.in,Expedited,JNE3371,JNE3371-KR-XL,kurta,XL,B07WV4JV4D,Shipped,1,329.00,NAVI MUMBAI,maharashtra,410210.0,IN,True
3,3,403-9615377-8133951,4/30/2022,Cancelled,Merchant,Amazon.in,Standard,J0341,J0341-DR-L,Western Dress,L,B099NRCT7B,Shipped,0,753.33,PUDUCHERRY,puducherry,605008.0,IN,False
4,4,407-1069790-7240320,4/30/2022,Shipped,Amazon,Amazon.in,Expedited,JNE3671,JNE3671-TU-XXXL,Top,3XL,B098714BZP,Shipped,1,574.00,CHENNAI,tamil_nadu,600073.0,IN,False
